File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if(strs.length === 0) return "";

    const first = strs[0];

    for(let i = 0; i < first.length; i++){
        const char = first[i];

        for(let j = 1; j < strs.length; j++){
            if(i >= strs[j].length || strs[j][i] !== char){
                return first.slice(0, i);
            }
        }
    }

    return first;
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        min_len = min(len(s) for s in strs)
        for i in range(min_len):
            curr_char = strs[0][i]

            for s in strs[1:]:
                if s[i] != curr_char:
                    return strs[0][:i]

        return strs[0][:min_len]
